pkg/provisioner/ironic/ironic.go

```diff
@@ -739,8 +739,11 @@ func (p *ironicProvisioner) setDirectDeployUpdateOptsForNode(ironicNode *nodes.N
 		"image_disk_format":   imageData.DiskFormat,
 	}
 	updater.
-		SetInstanceInfoOpts(optValues, ironicNode).
-		SetTopLevelOpt("deploy_interface", "direct", ironicNode.DeployInterface)
+		SetInstanceInfoOpts(optValues, ironicNode)
+
+	if ironicNode.DeployInterface != "" {
```

Comment:
It's never "", you need to check for values "ramdisk" and "custom-agent" which belong to different deploy methods.

Reply:
Even after we delete it, it still has a value?